### PR TITLE
Conversion To TypeScript

### PR DESCRIPTION
**Pull request by 22patriciaaz22**
We have finished converting all present js files to typescript, and they now (finally) all compile without errors in code or function.

### CODE REVIEW

**Comment by strega-nil** [comment]
cc @dan-shaw

File: .prettierignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 assets
 .vscode
 node_modules
-python_scripts
\ No newline at end of file
+python_scripts
+dist
```
Comment (strega-nil) [inline .prettierignore]:
```suggestion
dist

```

File: ts/common.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+function copyCodePanel(id) {
+    let temp: any = document.getElementById(id);
```
Comment (strega-nil) [inline ts/common.ts]: I don't like the use of `any` here; can we try checking for the existence of `textContent`, and if `textContent` doesn't exist we throw?

File: ts/common.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    temp.value = temp.textContent;
+    temp.select();
+    document.execCommand('copy');
+    clearSelection();
+}
+
+// remove the highlight from selected text
+function clearSelection() {
+    let doc = document as any;
```
Comment (strega-nil) [inline ts/common.ts]: Same thing as above.

File: ts/gettingStarted.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+    $('.gs-tab-btn').click(function () {
+        var id = $(this).attr('id');
+        clickGetStartedTab(id.substring(7));
+    });
+});
+
+function clickGetStartedTab(platform) {
+    os = platform;
+    let elems: any = document.getElementsByClassName(os);
```
Comment (strega-nil) [inline ts/gettingStarted.ts]: Same question as above. Also, if we need to use `any`, could we do explicit casts?

File: ts/gettingStarted.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        else copyCodePanel('unix-' + step);
+    });
+
+    $('.gs-tab-btn').click(function () {
+        var id = $(this).attr('id');
+        clickGetStartedTab(id.substring(7));
+    });
+});
+
+function clickGetStartedTab(platform) {
```
Comment (strega-nil) [inline ts/gettingStarted.ts]: We probably want to have a type alias for `Platform`, something like:

```
type Platform = 'windows' | 'unix' | 'mac' | 'linux';
function clickGetStartedTab(platform : Platform) {
```

**Review by strega-nil — APPROVED** [review]
I think this is fine, there's just some minor changes; we can make the types better over time. The most important thing is flipping into typescript for now.

**Review by dan-shaw — APPROVED** [review]
I'm not too familiar with TypeScript, would it be possible to add types to functions and function arguments?

**Comment by strega-nil** [comment]
@dan-shaw it would, but also we can add them over time, ya know?

**Comment by dan-shaw** [comment]
That makes sense, otherwise there would be too many changes

**Review by dan-shaw — APPROVED** [review]

File: ts/common.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+// remove the highlight from selected text
+function clearSelection() {
+    if (window.getSelection) {
+        window.getSelection().removeAllRanges();
+    }
+}
+
+// determine what OS the user is on, used to render corresponding package installation code
+// default to Unix unless the user is on a Windows device
+function detectOS() {
```
Comment (strega-nil) [inline ts/common.ts]:
```suggestion
function detectOS(): Platform {
```

File: ts/gettingStarted.ts [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        if (step === 'step5') {
+            // the code for windows and unix is shared in this case
+            copyCodePanel('all-' + step);
+        } else if (os === 'windows') copyCodePanel('windows-' + step);
+        else copyCodePanel('unix-' + step);
+    });
+
+    $('.gs-tab-btn').click(function () {
+        var id = $(this).attr('id');
+        clickGetStartedTab(id.substring(7) as Platform);
```
Comment (strega-nil) [inline ts/gettingStarted.ts]: This probably doesn't work like you want? You probably want something like `id.split('-')[0]`?

**Review by strega-nil — APPROVED** [review]
There are some minor changes I'd like to see but otherwise, looks good :)

**Pull request merged**